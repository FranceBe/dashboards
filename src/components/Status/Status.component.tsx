import SignalCellularAltIcon from '@material-ui/icons/SignalCellularAlt'
import SignalCellularOffIcon from '@material-ui/icons/SignalCellularOff'
import { StatusContainer } from 'components/Status/status.style'
import React from 'react'
import { StatusProps } from 'types/status'

export const Status: React.FC<StatusProps> = ({ status }) => {
  const statusIcon =
    status === 'disconnected' ? (
      <SignalCellularOffIcon data-testid={'signal-off'} />
    ) : (
      <SignalCellularAltIcon data-testid={'signal-on'} />
    )

  return (
    <StatusContainer data-testid={'status-container'} status={status}>
      {statusIcon}
    </StatusContainer>
  )
}

export default Status
