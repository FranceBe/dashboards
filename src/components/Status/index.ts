import StatusComponent, { Status } from 'components/Status/Status.component'

export { Status }

export default StatusComponent
