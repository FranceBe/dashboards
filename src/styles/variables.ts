// Colors to use matching the graphic chart
export const palette: { [key: string]: string } = {
  blue_light: '#f0f0fc',
  green_primary: '#65db25',
  grey_primary: '#343536',
  red_primary: '#eb4034',
}

// Spaces to use to respect an 8px grid
export const spaces: { [key: string]: string } = {
  large: '24px',
  medium: '16px',
  regular: '8px',
  small: '4px',
}

// Font sizes to use in project
export const fontSizes: { [key: string]: string } = {
  small: '12px',
}
