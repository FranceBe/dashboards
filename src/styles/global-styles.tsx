import { createGlobalStyle } from 'styled-components'
import { palette } from 'styles/variables'

export const GlobalStyle = createGlobalStyle`
  body {
    margin: 1.5em;
    background-color: ${palette.blue_light};
    color: ${palette.grey_primary};
  }
  a {
    text-decoration: none;
    :active, :visited {
      color: ${palette.blue_primary};
    }
    :hover, :visited:hover {
      color: ${palette.blue_secondary};
    }
  }
`
