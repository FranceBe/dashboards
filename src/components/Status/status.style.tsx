import React from 'react'
import styled from 'styled-components'
import { fontSizes, palette, spaces } from 'styles/variables'

export const StatusContainer = styled((props) => <div {...props} />)<{
  status: 'connected' | 'disconnected'
}>`
  background-color: ${(props) =>
    props.status === 'disconnected' ? palette.red_primary : palette.green_primary};
  width: fit-content;
  border-radius: ${spaces.small};
  padding: ${spaces.regular};
  font-size: ${fontSizes.small};
`
