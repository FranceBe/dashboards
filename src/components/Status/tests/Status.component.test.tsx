import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { render, screen } from '@testing-library/react'
import Status from 'components/Status'
import React from 'react'
import { palette } from 'styles/variables'

describe('Status', () => {
  describe('Given status is "connected"', () => {
    it('should match snapshot', () => {
      const { container } = render(<Status status={'connected'} />)
      expect(container.firstChild).toMatchSnapshot()
    })
    it('should have green_primary background color', () => {
      const { container } = render(<Status status={'connected'} />)
      expect(container.firstChild).toHaveStyleRule('background-color', palette.green_primary)
    })
    it('should find the connected icon', () => {
      render(<Status status={'connected'} />)
      expect(screen.getByTestId('signal-on')).toBeInTheDocument()
    })
  })
  describe('Given status is "disconnected"', () => {
    it('should match snapshot', () => {
      const { container } = render(<Status status={'disconnected'} />)
      expect(container.firstChild).toMatchSnapshot()
    })
    it('should have red_primary background color', () => {
      const { container } = render(<Status status={'disconnected'} />)
      expect(container.firstChild).toHaveStyleRule('background-color', palette.red_primary)
    })
    it('should find the disconnected icon', () => {
      render(<Status status={'disconnected'} />)
      expect(screen.getByTestId('signal-off')).toBeInTheDocument()
    })
  })
})
