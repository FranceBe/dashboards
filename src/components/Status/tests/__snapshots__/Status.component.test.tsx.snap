// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Given status is "connected" should match snapshot 1`] = `
.c0 {
  background-color: #00C9A3;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 4px;
  padding: 8px;
  font-size: 0.75rem;
  box-shadow: 0 5px 25px 0 rgba(0,0,45,0.1),0 2px 6px 0 rgba(0,0,0,0.04), 0 16px 24px 0 rgba(0,0,46,0.06);
}

<div
  class="c0"
  data-testid="status-container"
  status="connected"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    data-testid="signal-on"
    focusable="false"
    title="connected"
    viewBox="0 0 24 24"
  >
    <path
      d="M2 22h20V2z"
    />
  </svg>
</div>
`;

exports[`Status Given status is "disconnected" should match snapshot 1`] = `
.c0 {
  background-color: #eb4034;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 4px;
  padding: 8px;
  font-size: 0.75rem;
  box-shadow: 0 5px 25px 0 rgba(0,0,45,0.1),0 2px 6px 0 rgba(0,0,0,0.04), 0 16px 24px 0 rgba(0,0,46,0.06);
}

<div
  class="c0"
  data-testid="status-container"
  status="disconnected"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    data-testid="signal-off"
    focusable="false"
    title="disconnected"
    viewBox="0 0 24 24"
  >
    <path
      d="M21 1l-8.59 8.59L21 18.18V1zM4.77 4.5L3.5 5.77l6.36 6.36L1 21h17.73l2 2L22 21.73 4.77 4.5z"
    />
  </svg>
</div>
`;
